packages/react-instantsearch/native.js

```diff
@@ -0,0 +1,5 @@
+import InstantSearch from './src/core/InstantSearch';
```

Comment:
Yes you're right, I can pass the algoliasearch as a HOC props.

In the meantime, do we need to keep this dual mode between algoliaClient with appid/appkey already provided, and the one that we instantiate ourself.
Can we just ask for the algoliasearch dependency and then do the algoliasearch(appId, appKey) all the time?